x-pack/plugins/ml/server/routes/system.ts

```diff
@@ -187,10 +187,15 @@ export function systemRoutes(
 
           let isMlAutoscalingEnabled = false;
           try {
-            await client.asInternalUser.autoscaling.getAutoscalingPolicy({ name: 'ml' });
+            // kibana_system user does not have the manage_autoscaling cluster privilege.
+            // perform this check as a current user.
+            await client.asCurrentUser.autoscaling.getAutoscalingPolicy({ name: 'ml' });
             isMlAutoscalingEnabled = true;
           } catch (e) {
-            // If doesn't exist, then keep the false
+            // If ml autoscaling policy doesn't exist or the user does not have privileges to fetch it,
+            // check the number of lazy ml nodes to determine if autoscaling is enabled.
+            const lazyMlNodeCount = await getLazyMlNodeCount(client);
```

Comment:
It seems like this check will always work and so we don't really need to attempt to fetch `getAutoscalingPolicy` as they'll produce the same results.